File: better_api/KutnoAPI/KutnoAPI/Models/FinancialSummary.cs
using System.Text.Json.Serialization;

namespace KutnoAPI.Models
{
    public class FinancialSummary
    {
        [JsonPropertyName("regon")]
        public string Regon { get; set; }

        [JsonPropertyName("year")]
        public int Year { get; set; }

        [JsonPropertyName("yearpart")]
        public int YearPart { get; set; }

        [JsonPropertyName("department")]
        public string Department { get; set; }

        [JsonPropertyName("chapter")]
        public string Chapter { get; set; }

        [JsonPropertyName("paragraph")]
        public int Paragraph { get; set; }

        [JsonPropertyName("plannedincome")]
        public decimal PlannedIncome { get; set; }

        [JsonPropertyName("receivables")]
        public decimal Receivables { get; set; }

        [JsonPropertyName("madeincome")]
        public decimal MadeIncome { get; set; }

        [JsonPropertyName("plannedcost")]
        public decimal PlannedCost { get; set; }

        [JsonPropertyName("engagement")]
        public decimal Engagement { get; set; }

        [JsonPropertyName("madecost")]
        public decimal MadeCost { get; set; }
    }
}

File: data_prep/FinancialSummary.py
import requests
from requests.packages.urllib3.exceptions import InsecureRequestWarning
import pandas as pd

# Disable SSL certificate verification warning (not recommended for production)
requests.packages.urllib3.disable_warnings(InsecureRequestWarning)

# url = "http://localhost:5286/uploadfinancialdata/sendbytes"
url = "http://146.59.35.43:8080/financialdata"

payload = """{
}"""
headers = {
  'Content-Type': 'application/json'
}

response = requests.request("POST", url, headers=headers, data=payload)

df = pd.read_json(response.text)
print(df)


aggregated_df = df.groupby("regon").agg({
    "year": "first",
    "plannedincome": "sum",
    "receivables": "sum",
    "madeincome": "sum",
    "plannedcost": "sum",
    "engagement": "sum",
    "madecost": "sum",
}).reset_index()

print(aggregated_df)
